src/sources/dnstap/parser.rs

```diff
@@ -0,0 +1,1053 @@
+                    IpAddr::V6(Ipv6Addr::from(address_buffer))
+                };
+
+                self.insert(
+                    self.event_schema.dnstap_message_schema.query_address,
+                    source_address.to_string(),
+                );
+            }
+
+            if let Some(query_port) = dnstap_message.query_port {
+                self.insert(
+                    self.event_schema.dnstap_message_schema.query_port,
+                    query_port as i64,
+                );
+            }
+
+            if let Some(response_address) = dnstap_message.response_address {
+                let response_addr = if socket_family == 1 {
+                    let address_buffer: [u8; 4] = response_address[0..4].try_into()?;
+                    IpAddr::V4(Ipv4Addr::from(address_buffer))
+                } else {
+                    let address_buffer: [u8; 16] = response_address[0..16].try_into()?;
+                    IpAddr::V6(Ipv6Addr::from(address_buffer))
+                };
+
+                self.insert(
+                    self.event_schema.dnstap_message_schema.response_address,
+                    response_addr.to_string(),
+                );
+            }
+
+            if let Some(response_port) = dnstap_message.response_port {
+                self.insert(
+                    self.event_schema.dnstap_message_schema.response_port,
+                    response_port as i64,
+                );
+            }
+        }
+
+        if let Some(query_zone) = dnstap_message.query_zone {
+            let mut decoder: BinDecoder = BinDecoder::new(&query_zone);
+            match Name::read(&mut decoder) {
+                Ok(raw_data) => {
+                    self.insert(
+                        self.event_schema.dnstap_message_schema.query_zone,
+                        raw_data.to_utf8(),
+                    );
+                }
+                Err(error) => return Err(Error::from(error.to_string())),
+            }
+        }
+
+        let dnstap_message_type_id = dnstap_message.r#type;
+        self.insert(
+            &self
+                .event_schema
+                .dnstap_message_schema
+                .dnstap_message_type_id,
+            dnstap_message_type_id as i64,
+        );
+
+        self.insert(
+            self.event_schema.dnstap_message_schema.dnstap_message_type,
+            format!("{:?}", to_dnstap_message_type(dnstap_message_type_id)?),
+        );
+
+        let request_message_key = self
+            .event_schema
+            .dnstap_message_schema
+            .request_message
+            .to_string();
+        let response_message_key = self
+            .event_schema
+            .dnstap_message_schema
+            .response_message
+            .to_string();
+
+        if let Some(query_time_sec) = dnstap_message.query_time_sec {
+            let mut timestamp_nanosec: i64 = query_time_sec as i64 * 1_000_000_000_i64;
```

Comment:
I would like to, but the reason of using `i64` instead of `u64` (or even `u128`) is that `Event::Value::Integer` uses `i64` as its data type.
Unless we are planning to upgrade `Event::Value::Integer` to use `i128` in the near future (like the next couple of weeks 😄 ), I guess we'd better to leave the variable as `i64` as before.

Reply:
Aha, true, fair enough 👍